ibis/expr/types.py

```diff
@@ -192,14 +194,34 @@ def execute(self, limit='default', params=None, **kwargs):
           Pass an integer to effect a specific row limit. limit=None means "no
           limit". The default is whatever is in ibis.options.
 
+        timecontext: Optional[TimeContext], default None.
+           TimeContext = Tuple[SupportsTimestamp, SupportsTimestamp]
+           SupportsTimestamp = Union[pd.Timestamp, str]
+
+           Defines a time range of [begin, end). When defined, the execution
+           will only compute result for data inside the time range. This is
+           conceptually same as a time filter.
+           The time column must be named as 'time' and should preserve
+           across the expression. e.g. If that column is dropped then
+           execute will result in an error.
         Returns
         -------
         result : expression-dependent
           Result of compiling expression and executing in backend
         """
         from ibis.client import execute
 
-        return execute(self, limit=limit, params=params, **kwargs)
+        # don't pass timecontext to kwargs if ont present
```

Comment:
why is this needed? i would always pass as other args (it certainly can have a None value)

Reply:
This is necessary because https://github.com/ibis-project/ibis/blob/master/ibis/omniscidb/client.py#L809 for example, `omniscidb` backend doesn't like extra `kwargs`. At first place I removed this key in kwargs in `omniscidb` but Li suggested that it's better to do it here.

Reply:
fix that backend, its not in compliance

Reply:
OK, I will change it back to fix that backend then.